src/compute/src/sink/persist_sink.rs

```diff
@@ -40,68 +49,209 @@ where
+///    only uploaded/prepared to be appended to a shard. Also: we only write
+///    updates for batch descriptions that we learned about from
+///    `mint_batch_descriptions`.
+/// 3. `append_batches` takes as input the minted batch descriptions and written
+///    batches. Whenever the frontiers sufficiently advance, we take a batch
+///    description and all the batches that belong to it and append it to the
+///    persist shard.
+fn install_desired_into_persist<G>(
+    sink_id: GlobalId,
+    target: &CollectionMetadata,
+    desired_collection: Collection<G, Result<Row, DataflowError>, Diff>,
+    persist_collection: Collection<G, Result<Row, DataflowError>, Diff>,
+    as_of: Antichain<Timestamp>,
+    compute_state: &mut crate::compute_state::ComputeState,
+) -> Option<Rc<dyn Any>>
+where
+    G: Scope<Timestamp = Timestamp>,
+{
     let persist_clients = Arc::clone(&compute_state.persist_clients);
-    let persist_location = target.persist_location.clone();
     let shard_id = target.data_shard;
 
-    let operator_name = format!("persist_sink({})", shard_id);
-    let mut persist_op = OperatorBuilder::new(operator_name, scope.clone());
-
-    // We want exactly one worker (in the cluster) to send all the data to persist. It's fine
-    // if other workers from replicated clusters write the same data, though. In the real
-    // implementation, we would use a storage client that transparently handles writing to
-    // multiple shards. One shard would then only be written to by one worker but we get
-    // parallelism from the sharding.
-    // TODO(aljoscha): Storage must learn to keep track of collections, which consist of
-    // multiple persist shards. Then we should set it up such that each worker can write to one
-    // shard.
-    let hashed_id = sink_id.hashed();
-    let active_write_worker = (hashed_id as usize) % scope.peers() == scope.index();
+    let operator_name = format!("persist_sink {}", sink_id);
 
-    let mut input = persist_op.new_input(&ok_collection.inner, Exchange::new(move |_| hashed_id));
-    let mut err_input =
-        persist_op.new_input(&err_collection.inner, Exchange::new(move |_| hashed_id));
+    if sink_id.is_user() {
+        trace!(
+            "persist_sink {sink_id}/{shard_id}: \
+            initial as_of: {:?}",
+            as_of
+        );
+    }
 
-    let token = Rc::new(());
-    let token_weak = Rc::downgrade(&token);
+    let mut scope = desired_collection.inner.scope();
+
+    // The append operator keeps capabilities that it downgrades to match the
+    // current upper frontier of the persist shard. This frontier can be
+    // observed on the persist_feedback_stream. This is used by the minter
+    // operator to learn about the current persist frontier, driving it's
+    // decisions on when to mint new batches.
+    let (persist_feedback_handle, persist_feedback_stream) = scope.feedback(Timestamp::default());
```

Comment:
might want to document why its safe to have the summary timestamp 0 here!

Reply:
adding :ok_hand: